csharp/extractor/Semmle.Extraction.CSharp/Entities/Statements/GlobalStatementsBlock.cs

```diff
@@ -0,0 +1,41 @@
+using Semmle.Extraction.Kinds;
+using System.Linq;
+using System.IO;
+using Semmle.Extraction.Entities;
+
+namespace Semmle.Extraction.CSharp.Entities.Statements
+{
+    internal class GlobalStatementsBlock : Statement
+    {
+        private GlobalStatementsBlock(Context cx, Method parent)
+            : base(cx, StmtKind.BLOCK, parent, 0) { }
+
+        public override Microsoft.CodeAnalysis.Location ReportingLocation
+        {
+            get
+            {
+                // We only create a `GlobalStatementsBlock` if there are global statements. This also means that the
+                // entry point is going to be the generated method around those global statements
+                return cx.Compilation.GetEntryPoint(System.Threading.CancellationToken.None)
+                    ?.DeclaringSyntaxReferences
+                    .FirstOrDefault()
+                    ?.GetSyntax()
+                    .GetLocation();
```

Comment:
This is probably returning the `Method` that's passed into the constructor as the parent.